Pope blogposts auf einem Server nachahmen.

const posts = [
    { title: 'Post One', body: 'This is post one' },
    { title: 'Post Two', body: 'This is post two' },
];

kan vom server par sekunden dauern, deshalb setTimeOut


function getPosts() {
    setTimeout(() => {
        let output = '';
        posts.forEach((post) => {
            output += `<li>${post.title}</li>`;
        });
        document.body.innerHTML = output;
    }, 1000);
}

function createPost(post) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            posts.push(post);

            const error = false;

            if (!error) {
                resolve();
            } else {
                reject('Error: Something went wrong');
            }
        }, 2000);
    });
}

createPost({ title: 'Post Three', body: 'This is post three' }).then(getPosts).catch(err => console.log(err)); 

sobald es resolved ruft es getPosts auf, error abfangen mit catch, damit man eine besser aussehende error ausgabe hat


promise.all anstatt immer .xy ganze zeit zu schreiben, createPost auskommentiren, damit der code unten funktioniert

const promise1 = Promise.resolve('Hello World');
const promise2 = 10;
const promise3 = new Promise((resolve, reject) => setTimeOut(resolve, 2000, 'Goodbye'));
const promise4 = fetch('https://jsonplaceholder.typicode.com/users').then(response => response.json()); // fÃ¼r fetch muss man zweilmal .then nutzen


Promise.all([promise1, promise2, promise3]).then(((values) => console.log(values)));



